internal/controllers/topology/cluster/structuredmerge/dryrun_test.go

```diff
@@ -0,0 +1,888 @@
+				fieldsV1: map[string]interface{}{
+					"f:spec": map[string]interface{}{
+						"f:atomicMap": map[string]interface{}{},
+					},
+				},
+				original: map[string]interface{}{
+					"spec": map[string]interface{}{
+						"atomicMap": map[string]interface{}{
+							"foo": "bar",
+						},
+					},
+				},
+				modified: map[string]interface{}{
+					"spec": map[string]interface{}{
+						"atomicMap": map[string]interface{}{
+							"foo": "bar",
+						},
+					},
+				},
+			},
+			wantHasChanges:     false,
+			wantHasSpecChanges: false,
+		},
+		{
+			name: "identifies changes in an atomic map",
+			ctx: &dryRunInput{
+				path: contract.Path{},
+				fieldsV1: map[string]interface{}{
+					"f:spec": map[string]interface{}{
+						"f:atomicMap": map[string]interface{}{},
+					},
+				},
+				original: map[string]interface{}{
+					"spec": map[string]interface{}{
+						"atomicMap": map[string]interface{}{
+							"foo": "bar",
+						},
+					},
+				},
+				modified: map[string]interface{}{
+					"spec": map[string]interface{}{
+						"atomicMap": map[string]interface{}{
+							"foo": "bar-changed",
+						},
+					},
+				},
+			},
+			wantHasChanges:     true,
+			wantHasSpecChanges: true,
+		},
+		{
+			name: "No changes on managed atomic list",
+			ctx: &dryRunInput{
+				path: contract.Path{},
+				fieldsV1: map[string]interface{}{
+					"f:spec": map[string]interface{}{
+						"f:atomicList": map[string]interface{}{},
+					},
+				},
+				original: map[string]interface{}{
+					"spec": map[string]interface{}{
+						"atomicList": []interface{}{
+							map[string]interface{}{
+								"foo": "bar",
+							},
+						},
+					},
+				},
+				modified: map[string]interface{}{
+					"spec": map[string]interface{}{
+						"atomicList": []interface{}{
+							map[string]interface{}{
+								"foo": "bar",
+							},
+						},
+					},
+				},
+			},
+			wantHasChanges:     false,
+			wantHasSpecChanges: false,
+		},
+		{
+			name: "Identifies changes on managed atomic list",
+			ctx: &dryRunInput{
+				path: contract.Path{},
+				fieldsV1: map[string]interface{}{
+					"f:spec": map[string]interface{}{
+						"f:atomicList": map[string]interface{}{},
+					},
+				},
+				original: map[string]interface{}{
+					"spec": map[string]interface{}{
+						"atomicList": []interface{}{
+							map[string]interface{}{
+								"foo": "bar",
+							},
+						},
+					},
+				},
+				modified: map[string]interface{}{
+					"spec": map[string]interface{}{
+						"atomicList": []interface{}{
+							map[string]interface{}{
+								"foo": "bar-changed",
+							},
+						},
+					},
+				},
+			},
+			wantHasChanges:     true,
+			wantHasSpecChanges: true,
+		},
+		{
+			name: "No changes on managed listMap",
+			ctx: &dryRunInput{
+				path: contract.Path{},
+				fieldsV1: map[string]interface{}{
+					"f:spec": map[string]interface{}{
+						"f:listMap": map[string]interface{}{
+							"k:{\"foo\":\"id1\"}": map[string]interface{}{
```

Comment:
Isn't this inconsistent with foo + bar keys in the next two lines?

If yes, let's fix + check why the test didn't fail

Reply:
This is correct
each item in the list has two fields, foo and bar, but only one is used as a key in the test case (foo). you can consider bar as a attribute

Reply:
I think it is correct. If I got it right the
`"k:{\"foo\":\"id1\"}"` defines the ownership for the map[string]interface entry where the value of `foo` is `=="id1"`.

the `f:foo` and `f:bar` then says: inside the element of the listMap which has `foo=="id"` I have the ownership over the keys `foo` and `bar`.

Reply:
Okay got it, thx!!